homeassistant/components/zha/core/channels/security.py

```diff
@@ -147,23 +147,23 @@ def _disarm(self, code: str):
             self.alarm_status = AceCluster.AlarmStatus.No_Alarm
         return zigbee_reply
 
-    def _arm_day(self, code: str) -> None:
+    def _arm_day(self, code: str):
```

Comment:
We should correct the typing here and not remove it.

Reply:
`zha` is excluded from type checking because it has many wrong type hints like this one: https://github.com/home-assistant/core/blob/dev/script/hassfest/mypy_config.py#L140

Correcting them is a large task which requires some domain understanding. I think absent type hints are better than incorrect ones.

Reply:
CC @Adminiuga Can you give a change suggestion here?

Reply:
Would this work? it really returns a coroutine 
```suggestion
    def _arm_day(self, code: str) -> awaitable:
```